File: vokimi_api/Src/Err.cs
namespace vokimi_api.Src
{
    public record struct Err(string Message)
    {
        public Err(Exception ex) : this(ex.Message) { }
        public Err(string prefix, Err err) : this(prefix + err.ToString()) { }
        public override string ToString() => Message;
        public static Err None => new(string.Empty);
        public bool NotNone() => this != None;
    }
}

File: vokimi_frontend/src/ts/Err.ts
export class Err {
    private readonly message: string;

    constructor(message: string);
    constructor(ex: Error);
    constructor(arg: string | Error) {
        if (typeof arg === 'string') {
            this.message = arg;
        } else {
            this.message = arg.message;
        }
    }

    toString(): string {
        return this.message;
    }

    static none(): Err {
        return new Err('');
    }

    notNone(): boolean {
        return this.message !== '';
    }
}
